### DISCUSSION TITLE
DNS forwarding 은 무엇이고, 강의에서 언급한 예시가 궁금합니다.

### DISCUSSION THREAD

**Discussion by yeonkkk** [woowacourse-study/2023-tal-study#3] [Q&A]
강의에서 "PC DNS 주소와 공유기 주소가 같다면, DNS forwarding 기능으로 DNS를 대행하기도 한다" 라고 설명하셨는데, 정확한 의미가 무엇인지 궁금합니다!

**Comment by kyY00n** [comment]
(정리)
공유기 자체는 ~~DNS 역할을 하지는 않지만~~ PC입장에서는 공유기가 DNS 역할을 하는 것처럼 보인다는 말같습니다.
- 공유기(라우터) 가 DNS 테이블을 가지고 있어, ISP의 DNS 서버로 포워딩을 해줄 수 있습니다.
- 따라서 PC는 직접 ISP의 DNS 주소를 직접 가지고 있지 않아도, 공유기(라우터)의 주소를 이용해 인터넷에 연결할 수 있다는 말같아요

(수정)
공유기는 내부네트워크 DNS 역할을 할 수 있습니다.

**Comment by zillionme** [comment]
DNS 포워딩 기능은 내부 네트워크에서 호스트 이름이나 도메인 이름으로 인터넷에 접속할 때, 공유기가 미리 설정한 외부 DNS 서버에 해당 호스트 이름이나 도메인 이름에 대한 IP 주소를 요청하고, 이를 받아 내부 네트워크로 전달하여 인터넷에 연결하는 것이 가능합니다.

공유기는 DNS 테이블을 가지고 있습니다. DNS(Domain Name System)은 인터넷에서 도메인 이름을 IP 주소로 변환하는 서비스를 제공하는데, 이를 위해 DNS 서버를 사용합니다. 공유기는 내부 네트워크에서 호스트 이름이나 도메인 이름을 IP 주소로 변환하기 위해 DNS 서버 역할을 수행하며, 이를 위해 DNS 테이블을 유지합니다.
공유기의 DNS 테이블은 일반적으로 내부 네트워크에 연결된 기기들의 호스트 이름과 IP 주소를 매핑한 정보를 포함합니다. 이를 통해 내부 네트워크에서 호스트 이름이나 도메인 이름으로 인터넷에 접속할 때, 공유기는 먼저 내부 DNS 테이블에서 호스트 이름 또는 도메인 이름에 대한 IP 주소를 찾아보고, 없을 경우 외부 DNS 서버에 요청하여 IP 주소를 받아오게 됩니다.
또한, 공유기는 DNS 캐시를 유지하기도 합니다. DNS 캐시는 이전에 검색한 호스트 이름 또는 도메인 이름에 대한 IP 주소 정보를 저장해두어, 다음에 해당 정보를 요청할 때 외부 DNS 서버에 다시 요청하지 않고 빠르게 처리할 수 있도록 도와줍니다. DNS 캐시는 일정 기간이 지나면 자동으로 삭제되므로, 일시적인 변경에 대한 대응이 가능합니다.

DNS 포워딩 기능은 내부 네트워크에서 호스트 이름이나 도메인 이름으로 인터넷에 접속할 때, 공유기가 미리 설정한 외부 DNS 서버에 해당 호스트 이름이나 도메인 이름에 대한 IP 주소를 요청하고, 이를 받아 내부 네트워크로 전달하여 인터넷에 연결하는 것이 가능합니다.
[11:52](https://woowacourse.slack.com/archives/C053P8T1XK5/p1682304752729469)
@네, 공유기는 DNS 테이블을 가지고 있습니다. DNS(Domain Name System)은 인터넷에서 도메인 이름을 IP 주소로 변환하는 서비스를 제공하는데, 이를 위해 DNS 서버를 사용합니다. 공유기는 내부 네트워크에서 호스트 이름이나 도메인 이름을 IP 주소로 변환하기 위해 DNS 서버 역할을 수행하며, 이를 위해 DNS 테이블을 유지합니다.
공유기의 DNS 테이블은 일반적으로 내부 네트워크에 연결된 기기들의 호스트 이름과 IP 주소를 매핑한 정보를 포함합니다. 이를 통해 내부 네트워크에서 호스트 이름이나 도메인 이름으로 인터넷에 접속할 때, 공유기는 먼저 내부 DNS 테이블에서 호스트 이름 또는 도메인 이름에 대한 IP 주소를 찾아보고, 없을 경우 외부 DNS 서버에 요청하여 IP 주소를 받아오게 됩니다.
또한, 공유기는 DNS 캐시를 유지하기도 합니다. DNS 캐시는 이전에 검색한 호스트 이름 또는 도메인 이름에 대한 IP 주소 정보를 저장해두어, 다음에 해당 정보를 요청할 때 외부 DNS 서버에 다시 요청하지 않고 빠르게 처리할 수 있도록 도와줍니다. DNS 캐시는 일정 기간이 지나면 자동으로 삭제되므로, 일시적인 변경에 대한 대응이 가능합니다.
